### PR TITLE
tests(key-change): add tests and docs related to MACI's key change

### PR DESCRIPTION
**Pull request by ctrlc03**
Currently, it is not clear how the key change mechanism works (or doesn't), hence this PR aims to provide some clarification on this front, by adding docs and test cases.

This is in reference to this: https://github.com/privacy-scaling-explorations/maci/issues/717

### CODE REVIEW

**Comment by baumstern** [comment]
Super cool. It's much needed.

File: cli/tests/keyChange.test.ts [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+        90,
+        25,
+        25,
+        INT_STATE_TREE_DEPTH,
+        MSG_BATCH_DEPTH,
+        MSG_TREE_DEPTH,
+        VOTE_OPTION_TREE_DEPTH,
+        coordinatorPubKey,
+      );
+      stateIndex = parseInt(await signup(keypair1.pubKey.serialize()));
```
Comment (baumstern) [inline cli/tests/keyChange.test.ts:91]:
```suggestion
      stateIndex = await signup(keypair1.pubKey.serialize());
```

nit: it's better to return `number` type from `signup()`

**Review by baumstern — CHANGES_REQUESTED** [review]
Great work! Just left a nit

File: cli/tests/keyChange.test.ts [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    });
+    const keypair1 = new Keypair();
+    const keypair2 = new Keypair();
+    const initialNonce = 1;
+    const initialVoteOption = 0;
+    const initialVoteAmount = 9;
+    const pollId = 0;
+    let stateIndex = 0;
+    const expectedTally = initialVoteAmount - 1;
+    const expectedPerVOteOptionTally = (initialVoteAmount - 1) ** 2;
```
[the remark this answers is not in the document]
Comment (ctrlc03) [reply cli/tests/keyChange.test.ts:75]: Could you please rephrase that? I'm afraid I didn't get it

[the remark this answers is not in the document]
Comment (kittybest) [reply cli/tests/keyChange.test.ts:75]: oh just wanna ask why the character 'o' is capitalized XD

[the remark this answers is not in the document]
Comment (ctrlc03) [reply cli/tests/keyChange.test.ts:75]: ohh it's a typo 😆

File: cli/tests/keyChange.test.ts (hunk 1 above)
Comment (ctrlc03) [reply cli/tests/keyChange.test.ts:91]: Can you please dismiss your change request as this is something not related to this PR? will amend in another PR

File: cli/tests/keyChange.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ctrlc03) [reply cli/tests/keyChange.test.ts:75]: fixed :)

**Review by baumstern — APPROVED** [review]

**Pull request merged**